HaxeManual/06-language-features.tex

```diff
@@ -391,12 +399,20 @@ \subsection{Variable capture}
 	\item Otherwise \expr{"none"} is returned.
 \end{itemize}
 
-It is also possible to use = to capture values which are further matched:
+It is also possible to use \expr{=} to capture values which are further matched:
 
 \haxe[firstline=32,lastline=36]{assets/PatternMatching.hx}
 
 Here, \expr{leafNode} is bound to \expr{Leaf("foo")} if the input matches that. In all other cases, \expr{myTree} itself is returned: \expr{case x} works similar to \expr{case _} in that it matches anything, but with an identifier name like \expr{x} it also binds the matched value to that variable.
 
+\since{4.0.0}
+
+If the compiler detects that the name of a ``catch-all'' variable is very similar to an enum constructor, it will warn about a potential typo. This can be avoided by prefixing the variable identifier with \expr{var}:
+
+\haxe[firstline=107,lastline=113]{assets/PatternMatching.hx}
+
+Any capture variable can be prefixed with \expr{var}, but in most cases it is unnecessary. Since it also circumvents the warning about potential typos, it is recommended to not use \expr{var} unless necessary.
```

Comment:
That a weird recommendation. If you explicitly use `var` you probably want to capture a var.

Reply:
I'm trying to say you shouldn't use `var` by default for capture variables. That way if you make a typo you are warned. If you get used to always capturing with `var` you might not get a useful warning.

Reply:
I don't get it, why would you want a warning if you want to capture a variable anyway? In that case it doesn't matter how you name it and how similar the name is to an enum constructor.

Reply:
Never mind, you are right. I was thinking about it the wrong way.